src/middlewared/middlewared/plugins/jail.py

```diff
@@ -572,7 +579,8 @@ def list_resource(self, resource, remote, want_cache, branch):
                     plugin[i] = elem if elem != "-" else None
 
                 if remote:
-                    pv = self.get_plugin_version(plugin[2])
+                    plugin_version = plugins_versions_data.get(plugin[2], {}).get('version', 'N/A')
+                    pv = [plugin_version, '1' if plugin_version != 'N/A' else plugin_version]
```

Comment:
@skarekrow does the last entry here represent patch number ? In the old implementation, we have hardcoded it to `1` and `N/A` if version can't be determined. For now i've copied the same logic

Reply:
The last entry? Are you talking about index 2?

Reply:
Nope, i am referring to the last entry of `pv`

Reply:
Also @skarekrow, @william-gr i would urge we change how we show output right now, it is something along these lines
```
[
        "MineOS ",
        " MineOS is a web interface to set up and manage Minecraft servers with a few mouse clicks.",
        "mineos",
        "",
        "False",
        "https://www.trueos.org/iocage-icons/mineos.png",
        "N/A",
        "N/A"
      ],
```
UI will have to adapt to api changes for this one as it's a job now, they can perhaps take care of this too in the same way. I propose we convert this to a dictionary for readability.

Reply:
Completely agree

Reply:
> Nope, i am referring to the last entry of `pv`

Yes that is the version. Though in recent versions the very last index shows the PLUGIN_INFO file.

Reply:
I'll change the output format in a new ticket and have it up asap

Reply:
> > Nope, i am referring to the last entry of `pv`
> 
> Yes that is the version. Though in recent versions the very last index shows the PLUGIN_INFO file.

I am confused @skarekrow. First entry is the actual plugin version and the second entry is a hardcoded default of `1`. What does `1` represent ?

Reply:
IIRC it's the plugin schema, which most are 1. Unless they are a 2. In which case they specify it.